StitchSamplePushNotificationApp/app/src/main/java/com/mongodb/example/pushnotification/MainActivity.java

```diff
@@ -0,0 +1,171 @@
+
+    private static final String TAG = "StitchPushNotification";
+
+    private StitchClient stitchClient;
+
+    private GCMPushClient pushClient;
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+        stitchClient =  new StitchClient(this, "STICH-APP-ID");
+
+        initLogin();
+
+        setContentView(R.layout.activity_main);
+    }
+
+    private void initLogin() {
+        stitchClient.getAuthProviders().addOnCompleteListener(new OnCompleteListener<AvailableAuthProviders>() {
+            @Override
+            public void onComplete(@NonNull final Task<AvailableAuthProviders> task) {
+                if (task.isSuccessful()) {
+                    AvailableAuthProviders info = task.getResult();
+
+                    if (info.hasAnonymous()) {
+                        stitchClient.logInWithProvider(new AnonymousAuthProvider()).addOnCompleteListener(new OnCompleteListener<Auth>() {
+                            @Override
+                            public void onComplete(@NonNull final Task<Auth> task) {
+                                if (task.isSuccessful()) {
+                                    initGCMClient();
+                                } else {
+                                    Log.e(TAG, "Error logging in anonymously", task.getException());
+                                }
+                            }
+                        });
+                    } else {
+                        Log.e(TAG, "Enable Anonymous Login", task.getException());
+                    }
+
+                } else {
+                    Log.e(TAG, "Error getting authentication info", task.getException());
+                }
+            }
+        });
+    }
+
+    private void initGCMClient() {
+        stitchClient.getPushProviders().addOnSuccessListener(new OnSuccessListener<AvailablePushProviders>() {
+            @Override
+            public void onSuccess(final AvailablePushProviders availablePushProviders) {
+                if (!availablePushProviders.hasGCM()) {
+                    return;
+                }
+                pushClient = (GCMPushClient) stitchClient.getPush().forProvider(availablePushProviders.getGCM());
+                pushClient.register().addOnCompleteListener(new OnCompleteListener<Void>() {
+                    @Override
+                    public void onComplete(@NonNull final Task<Void> task) {
+                        if (!task.isSuccessful()) {
+                            Log.d(TAG, "Registration failed: " + task.getException());
+                            return;
+                        }
+                        Log.d(TAG, "Registration completed");
+                    }
+                });
+            }
+        });
+    }
+
+
+    /** Called when the user registers for push notification topics
+     * @param view
+     */
+    public void subscribeToTopic(View view) {
```

Comment:
Should add a comment here noting that this button is the `onClick` method for `button2`. Setting `onClick` in the XML, while convenient (and totally fine), is a bit of an antipattern, so it's always nice to give the developer some breadcrumbs in the actual Java file as to why this method exists.